File: 分类顺序表/1. Array/LC26. Remove Duplicates from Sorted Array/solution.py
class Solution(object):
    def removeDuplicates(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        i, j = 0,1
        for j in range(len(nums)):
            if(nums[j] != nums[i]):
                i += 1
                nums[i] = nums[j]
        
        return i+1
                

File: 分类顺序表/1. Array/LC26. Remove Duplicates from Sorted Array/solution.java
//solution1:slow runner & fast runner
class Solution {
    public int removeDuplicates(int[] nums) {
        if(nums.length == 0) return 0;
        
        int i = 0;
        for(int j = 1; j < nums.length; j++){
            if(nums[j] != nums[i]){
                nums[++i] = nums[j];
            }
        }
        
        return i+1;
    }
}

//time O(n), each of i and j traverses at most n steps
//space O(1), in place.
